src/components/report-viewer/text-filter-button.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, { ChangeEvent, Dispatch, FunctionComponent, SetStateAction, useState } from 'react';
+import { Box, IconButton, TextField } from '@mui/material';
+import FilterAltIcon from '@mui/icons-material/FilterAlt';
+import { useIntl } from 'react-intl';
+import Menu from '@mui/material/Menu';
+
+const styles = {
+    container: {
+        position: 'relative',
+    },
+    icon: {
+        width: '0.7em',
+        height: '0.7em',
+    },
+    notificationDot: {
+        height: '6px',
+        width: '6px',
+        backgroundColor: '#cc70a0',
+        borderRadius: '50%',
+        position: 'absolute',
+        top: '5px',
+        left: '23px',
+    },
+};
+
+/**
+ * FilterButton wraps a MultiSelectList with a button which has a visual indication to indicate when the user alters the default state of the list
+ * @param {String} filterText - Value of the filter
+ * @param {Function} setFilterText - Setter needed to update the filter
+ */
+
+interface TextFilterButtonProps {
+    filterText: string;
+    setFilterText: Dispatch<SetStateAction<string>>;
```

Comment:
```suggestion
    setFilterText: (text: string) => void;
```
This is enough from TextFilterButton perspective